src/lightning/fabric/fabric.py

```diff
@@ -1063,3 +1066,12 @@ def _configure_callbacks(callbacks: Optional[Union[List[Any], Any]]) -> List[Any
         callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
         callbacks.extend(_load_external_callbacks("lightning.fabric.callbacks_factory"))
         return callbacks
+
+
+def _update_device_attribute(module: torch.nn.Module, device: torch.device) -> None:
+    # NOTE: for sharded strategies or manual device placement, there's no single root device
+    if not isinstance(module, _DeviceDtypeModuleMixin):
+        return
+    # cannot use `module.to()` because we don't actually want to move the model in case there are multiple
+    # devices types (such as partial meta parameters)
+    module._device = device
```

Comment:
This won't propagate to submodules, while the previous way of calling .to() did. Can you incorporate that?

Reply:
Good point. Done